tox.ini

```diff
@@ -16,9 +16,15 @@ commands =
 basepython = python3.8
 deps =
     {[testenv]deps}
-    Scrapy==2.7.0
+    Scrapy==2.11.0
     url-matcher==0.5.0
-    w3lib==1.22.0
+    w3lib==2.0.1
+
+[testenv:rules]
+basepython = python3.12
+deps =
+    {[testenv:pinned]deps}
```

Comment:
Could we remove pinned versions from this one? Or do you want to specifically test the rules extra with pinned versions of dependencies?

Reply:
In that case, Python 3.8 made sense, Python 3.12 is better tested without pinned versions.

Reply:
Got it. Reverted. 👍